core/src/main/python/synapse/ml/core/platform/Platform.py

```diff
@@ -67,3 +67,9 @@ def find_secret(secret_name, keyvault=SECRET_STORE, override=None):
             f"and would like to manually specify your key for Azure KeyVault or Databricks Secrets,"
             f'please add the override="YOUR_KEY_HERE" to the arguments of the find_secret() method'
         )
+
+
+def display_for_synapse_batch_mode(data):
+    if isinstance(data, DataFrame):
+        data.collect()
```

Comment:
Replace the print with a call to the original display to preserve as much semantics as possible, other things in notebooks other than dfs can be displayed